settings/src/lib.rs

```diff
@@ -99,41 +101,84 @@ impl Default for PaymentSettings {
+    #[serde(default)]
+    pub state: ExitState,
+    #[serde(default)]
+    pub message: String,
+}
+
 #[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq)]
 pub struct ExitClientSettings {
-    pub exit_ip: IpAddr,
-    pub exit_registration_port: u16,
+    #[serde(default)]
+    pub exits: HashMap<String, ExitServer>,
+    pub current_exit: Option<String>,
     pub wg_listen_port: u16,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub details: Option<ExitClientDetails>,
-    pub reg_details: ExitRegistrationDetails,
+    pub reg_details: Option<ExitRegistrationDetails>,
 }
 
 impl Default for ExitClientSettings {
     fn default() -> Self {
         ExitClientSettings {
-            exit_ip: "fd00::8".parse().unwrap(),
-            exit_registration_port: 4875,
+            exits: HashMap::new(),
+            current_exit: None,
             wg_listen_port: 59999,
-            details: None,
-            reg_details: ExitRegistrationDetails {
+            reg_details: Some(ExitRegistrationDetails {
                 zip_code: Some("1234".into()),
                 email: Some("1234@gmail.com".into()),
                 country: Some("Althea".into()),
-            },
+            }),
         }
     }
 }
 
-#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq)]
-pub struct ExitClientDetails {
-    pub own_internal_ip: IpAddr,
-    pub server_internal_ip: IpAddr,
-    pub netmask: u8,
-    pub eth_address: EthAddress,
-    pub wg_public_key: String,
-    pub wg_exit_port: u16,
-    pub exit_price: u64,
+impl ExitClientSettings {
+    pub fn get_current_exit(&self) -> Option<&ExitServer> {
+        Some(&self.exits[self.current_exit.as_ref()?])
+    }
+    pub fn get_current_exit_name(&self) -> Option<&ExitServer> {
+        Some(&self.exits[self.current_exit.as_ref()?])
+    }
+    pub fn get_current_exit_details(&self) -> Option<&ExitClientDetails> {
+        match self.get_current_exit() {
+            Some(ref exit) => match exit.our_details {
+                Some(ref details) => return Some(details),
+                _ => (),
+            },
+            _ => (),
+        }
+        None
+    }
+    pub fn set_current_exit(&mut self) -> Option<&mut ExitServer> {
+        self.exits.get_mut(self.current_exit.as_ref()?)
+    }
+    pub fn current_exit_exists(&self) -> bool {
+        if self.current_exit == None {
+            return false;
+        }
+        return self.exits.contains_key(self.current_exit.as_ref().unwrap());
+    }
+    pub fn current_exit_details_exists(&self) -> bool {
+        if self.current_exit == None {
+            return false;
+        }
+        if self.exits.contains_key(self.current_exit.as_ref().unwrap()) == false {
+            return false;
+        }
+        return self.exits[self.current_exit.as_ref().unwrap()]
+            .our_details
+            .is_some();
+    }
+    pub fn reg_details_set(&self) -> bool {
+        self.reg_details.is_some()
```

Comment:
Writing the body of this function takes literally 4 more letters. Why does it exist? Also, why the getters and setters for all the different fields, but not `reg_details`?